subsys/bluetooth/services/fast_pair/fp_storage/CMakeLists.txt

```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-add_library(fp_storage STATIC)
```

Comment:
There was a particular reason why this library is added this way (@MarekPieta probably remembers it better). See this PR: https://github.com/nrfconnect/sdk-nrf/pull/8131 and especially this comment: https://github.com/nrfconnect/sdk-nrf/pull/8131#discussion_r916012276. Basically now we are mixing zephyr commands with cmake commands. Particularly under `zephyr_library_named` we do `target_link_libraries(fp_storage PUBLIC zephyr_interface)` and in this file in line 31 we have `target_link_libraries(fp_storage PRIVATE zephyr_interface)`. Everything seems to be working fine, but there might be some hidden consequences which I don't see/understand.

Reply:
> There was a particular reason why this library is added this way (@MarekPieta probably remembers it better). See this PR: #8131 and especially this comment: [#8131 (comment)](https://github.com/nrfconnect/sdk-nrf/pull/8131#discussion_r916012276) (`fp_crypto` and `fp_storage` libraries are created in similar way). Basically now we are mixing zephyr commands with cmake commands. Particularly under `zephyr_library_named` we do `target_link_libraries(fp_storage PUBLIC zephyr_interface)` and in this file in line 31 we have `target_link_libraries(fp_storage PRIVATE zephyr_interface)`. Everything seems to be working fine, but there might be some hidden consequences which I don't see/understand.

Reply:
It will work if you do:
`add_library(fp_storage STATIC)
zephyr_append_cmake_library(fp_storage)`
It is clearer and don't duplicate `target_link_libraries` call.

Reply:
Thanks for pointing this out. It seems that something has changed since we added `fp_storage` library.
After applying my minor suggestions, the PR should be fine and ready for merging :)

Reply:
aligned also the crypto part